src/Mesh.h

```diff
@@ -25,19 +25,43 @@ class Mesh
     public:
     Mesh(const std::string filename);
 
+    /**
+     * Get the physical x or y coordinate of an element node on the host Kokkos::View.
+     */
     [[nodiscard]]
-    inline double GetCoordinate(int element, int node, int dim) {return data_(element,node,dim+1);}
+    inline double GetCoordinate(int element, int node, int dim) const {return data_(element,node,dim+1);}
 
+    /**
+     * Get the global degree of freedom on the host Kokkos::View.
+     */
     [[nodiscard]]
-    inline int GetGlobalDof(int element, int node) {return data_(element,node,0);}
+    inline int GetGlobalDof(int element, int node) const {return data_(element,node,0);}
+
+    #ifdef KOKKOS_ENABLE_CUDA
+        /**
+         * Get the physical x or y coordinate of an element node on the device Kokkos::View.
+         */
+        [[nodiscard]] 
+        inline double GetDeviceCoordinate(int element, int node, int dim) const {return device_data_(element,node,dim+1);}
+        /**
+         * Get the global degree of freedom (DOF) number on the device Kokkos::View.
+         */
+        [[nodiscard]]
+        inline int GetDeviceGlobalDof(int element, int node) const {return device_data_(element,node,0);}
+    #endif
```

Comment:
Instead of using two different versions of `GetCoordinates` function, use the `KOKKOS_INLINE` keyword to mark it device or host function automatically. Here, you will find examples: https://kokkos.org/kokkos-core-wiki/ProgrammingGuide/ParallelDispatch.html